base/ca/src/com/netscape/cms/servlet/cert/scep/CRSEnrollment.java

```diff
@@ -610,6 +614,60 @@ public void handleGetCACert(HttpServletRequest httpReq,
 
     }
 
+    /**
+     * Returns the CA capabilities in a series of plaintext lines.
+     *
+     * @param httpReq The HttpServletRequest.
+     * @param httpResp The HttpServletResponse.
+     *
+     */
+
+    public void handleGetCACaps(HttpServletRequest httpReq,
+                              HttpServletResponse httpResp)
+            throws ServletException {
+
+        try {
+            StringBuilder response = new StringBuilder();
+
+            /*
+             Possible capabilities as of https://tools.ietf.org/html/draft-gutmann-scep-16#section-3.5.1:
+             - AES (currently not supported by Dogtag)
+             - DES3
+             - GetNextCACert (currently not supported by Dogtag)
+             - POSTPKIOperation (currently not supported by Dogtag)
+             - Renewal (currently not supported by Dogtag)
+             - SHA-1
+             - SHA-256
+             - SHA-512
+             - SCEPStandard (currently not supported by Dogtag due to missing AES support)
+            */
+            if (isAlgorithmAllowed(mAllowedEncryptionAlgorithm, "DES3")) {
+                response.append("DES3\n");
+            }
+            // response.append("POSTPKIOperation\n");
+            if (isAlgorithmAllowed(mAllowedHashAlgorithm, "SHA1")) {
+                response.append("SHA-1\n");
+            }
+            if (isAlgorithmAllowed(mAllowedHashAlgorithm, "SHA256")) {
+                response.append("SHA-256\n");
+            }
+            if (isAlgorithmAllowed(mAllowedHashAlgorithm, "SHA512")) {
+                response.append("SHA-512\n");
+            }
+
+            httpResp.setContentType("text/plain");
+            httpResp.setContentLength(response.length());
+            httpResp.getOutputStream().print(response.toString());
+            httpResp.getOutputStream().flush();
+
+            logger.debug("Output CA Capabilities:");
+            logger.debug(response.toString());
+        } catch (Exception e) {
+            logger.error("CRSEnrollment: failed sending CA capabilities", e);
+            throw new ServletException("Failed sending CA capabilities");
```

Comment:
I'd suggest chaining the original exception whenever possible to help troubleshooting:
```
throw new ServletException("Failed sending CA capabilities: " + e.getMessage(), e);
```